numpy/lib/tests/test_function_base.py

```diff
@@ -4510,3 +4510,33 @@ def test_sort_complex(self):
         actual = np.sort_complex(a)
         assert_equal(actual, expected)
         assert_equal(actual.dtype, expected.dtype)
+
+
+def test_quantile_zero_weights():
+    """
+    Test np.quantile raises an error when weights are all zeros.
+    (Corresponds to the example given in bug #28589)
+    """
+
+    arr = np.array([1, 2, 3, 4])
+    quantile = 0.5
+    weights = np.array([0, 0, 0, 0])
+
+    with pytest.raises(ValueError, match="All weights are zero, cannot compute quantile."):
+        np.quantile(arr, quantile, weights=weights, method='inverted_cdf')
+
+
+def test_quantile_valid_weights():
+    """
+    Test np.quantile with valid weights.
+    """
+    arr = np.array([1, 2, 3, 4])
+    quantile = 0.5
+    weights = np.array([1, 1, 1, 1])
+
+    result = np.quantile(arr, quantile, weights=weights, method='inverted_cdf')
+    expected = 2.5
+
+    assert np.isclose(result, expected)
```

Comment:
Hopefully this part of the diff isn't really needed--presumably we have sufficient testing that if you broke the numerical behavior of `quantile` an existing test would fail.

That said, this test actually does fail--maybe there is a problem with the test design?

Reply:
As an aside, `assert_allclose()` is generally preferable in testing--it tends to give better feedback when something fails, etc.